optimum_benchmark/benchmarks/training.py

```diff
@@ -1,31 +1,30 @@
-from typing import Any, Optional, Dict, TYPE_CHECKING
+from typing import Any, Dict
 from dataclasses import dataclass, field
 from logging import getLogger
 
-from transformers import default_data_collator
 from omegaconf import OmegaConf
 from pandas import DataFrame
-import torch
 
-from optimum_benchmark.benchmarks.base import Benchmark, BenchmarkConfig
-from optimum_benchmark.generators.dataset_generator import DatasetGenerator
-
-if TYPE_CHECKING:
-    from optimum_benchmark.backends.base import Backend
+from ..backends.base import Backend
+from .base import Benchmark, BenchmarkConfig
+from ..generators.dataset_generator import DatasetGenerator
+from .training_utils import MeasurementCallback, get_data_collator
 
 
 LOGGER = getLogger("training")
 
 # resolvers
 OmegaConf.register_new_resolver("is_cpu", lambda device: device == "cpu")
-OmegaConf.register_new_resolver("device_count", lambda: torch.cuda.device_count())
 
 
 @dataclass
 class TrainingConfig(BenchmarkConfig):
     name: str = "training"
     _target_: str = "optimum_benchmark.benchmarks.training.TrainingBenchmark"
 
+    # training options
+    warmup_steps: int = 2
```

Comment:
This is much too low. Can we leave a higher default? Also, changing defaults like this is breaking older config that are not greedy (not saving all params), I think it can be an issue.

See https://github.com/huggingface/optimum-benchmark/pull/28

Reply:
depends on the hardware but we can keep it 40.

Reply:
Yes agree, it probably depend on the hardware / model / batch size / etc.